spark/v3.1/spark/src/main/scala/com/netease/arctic/spark/sql/execution/ExtendedArcticUnkeyedStrategy.scala

```diff
@@ -29,17 +28,24 @@ import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog}
 import org.apache.spark.sql.connector.iceberg.read.SupportsFileFilter
 import org.apache.spark.sql.execution.datasources.v2._
 import org.apache.spark.sql.execution.{FilterExec, LeafExecNode, ProjectExec, SparkPlan}
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
 import org.apache.spark.sql.{SparkSession, Strategy}
 
+import scala.collection.JavaConverters
 import scala.collection.JavaConverters.seqAsJavaList
 
 case class ExtendedArcticUnkeyedStrategy(spark: SparkSession) extends Strategy {
   override def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
-    case CreateArcticTableStatement(ArcticCatalogAndIdentifier(catalog, identifier),
-    schema, partitioning, _, properties, _, _, _, _, _, primary, _, ifNotExists) =>
-      CreateArcticTableStatementExec(catalog, identifier, schema,
-        partitioning, properties, seqAsJavaList(primary), ifNotExists) :: Nil
-
+    case CreateTableAsSelect(catalog, ident, parts, query, props, options, ifNotExists) =>
+      var propertiesMap: Map[String, String] = props
+      var optionsMap: Map[String, String] = options
+      if (options.contains("primary.keys")) {
+        propertiesMap += ("primary.keys" -> options("primary.keys"))
+        optionsMap += ("overwrite-mode" -> "dynamic")
+      }
+      val writeOptions = new CaseInsensitiveStringMap(JavaConverters.mapAsJavaMap(optionsMap))
+      CreateTableAsSelectExec(catalog, ident, parts, query, planLater(query),
+        propertiesMap, writeOptions, ifNotExists) :: Nil
```

Comment:
case CreateTableAsSelect() if isArcticTable && isHasPrimaryKey => 
    // code

only handle command run on **arctic  primary key** table

Reply:
Could not identify whether it is an arctic table, `options.contains("primary.keys")` can prove ` isHasPrimaryKey`.
This code is compatible with the original CTAS